metabase/metabase
# Show new syncing tables as greyed out in data reference pane

Closes #32591

### Description

Disables and grays out links to not-fully-synced tables in native query editor's right-hand-side sidebar.

Design doc: https://www.notion.so/metabase/Provide-progress-on-initial-database-sync-16af8e79312a45ecb715d8e58801c52b#5d67f984521843c4818e17a37719a111

#### Before

![image](https://github.com/metabase/metabase/assets/6830683/080b96f0-313e-4a46-a034-97712f591037)

#### After

![image](https://github.com/metabase/metabase/assets/6830683/b0e14ec9-dd53-4592-b1ed-ca34ed00b339)

### How to verify

1. Simulate long sync with this patch (as shown in [this video](https://www.loom.com/share/e3e9b5a2c2a441aaba4c94ca4f01b776) from [this PR](https://github.com/metabase/metabase/pull/31609)) - _make sure to rebuild_:
   ```diff
   diff --git a/src/metabase/sync/sync_metadata/fields.clj b/src/metabase/sync/sync_metadata/fields.clj
   index 58147f61eb..55afecc5b0 100644
   --- a/src/metabase/sync/sync_metadata/fields.clj
   +++ b/src/metabase/sync/sync_metadata/fields.clj
   @@ -80,6 +80,7 @@
                                    :total-fields   su/IntGreaterThanOrEqualToZero})
      "Sync the Fields in the Metabase application database for all the Tables in a `database`."
      [database :- i/DatabaseInstance]
   +  (Thread/sleep 30000)
      (->> database
         sync-util/db->sync-tables
         (map (partial sync-fields-for-table! database))
   ```
2. Spin up the QA dbs:
    ```Shell
    yarn test-qa-dbs:up
    ````
3. Go to http://localhost:3000/admin/databases/create and connect Postgres QA DB ([credentials](https://github.com/metabase/metabase/blob/d6473cdce794254293e4be2cc4f17aef527ccf05/e2e/support/cypress_data.js#L164-L170)).
4. Now, within 30 seconds, click: "**+ New**" > "**SQL query**" > "**Learn more about your data**" > "**{name-of-the database-connected-in-the-previous-step}**"
5. You should see a list of 4 tables in that database.

Expected result: the tables are grayed out and non-interactive.

### Demo

https://github.com/metabase/metabase/assets/6830683/9e3ec84d-1e87-4e5e-b02e-a9acd590719e
